src/onnx/parse_scatter.cpp

```diff
@@ -0,0 +1,43 @@
+#include <migraphx/onnx/op_parser.hpp>
+#include <migraphx/instruction.hpp>
+#include <migraphx/ranges.hpp>
+#include <migraphx/make_op.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace onnx {
+
+struct parse_scatter : op_parser<parse_scatter>
+{
+    std::vector<op_desc> operators() const { return {{"ScatterElements"}}; }
+
+    instruction_ref parse(const op_desc& /*opd*/,
+                          const onnx_parser& /*parser*/,
+                          const onnx_parser::node_info& info,
+                          const std::vector<instruction_ref>& args) const
+    {
+        operation op;
+
+        std::string op_name = "scatter_none";
+        int axis            = 0;
+
+        if(contains(info.attributes, "axis"))
+            axis = info.attributes.at("axis").i();
+        if(contains(info.attributes, "reduction"))
+        {
+            if(info.attributes.at("reduction").s() == "add")
```

Comment:
Referring to the comment above, I would assign the `reduction_mode` variable to `info.attributes.at("reduction"_.s())`.  This way you can just have a single conditional statement that checks if the string is a valid attribute.

So something like this:
```suggestion
            reduction_mode = info.attributes.at("reduction").s();
            if(reduction_mode != "none" or reduction_mode != "add" or reduction_mode != "mul")
                MIGRAPHX_THROW("PARSE_SCATTER: unsupported reduction mode " + reduction_mode);
```

Reply:
discussed this at length with Paul and concluded it's not appropriate to throw an error here.  The Onnx docs don't say an unsupported reduction attribute is an error; they just say that "none" is the default, so that's what I did.

Reply:
> We can still throw an error. I, personally, didnt think it was important, but its not bad to have some better mesages when something goes wrong.
> 
> However, I would suggest checking the error with `if(not contains({"none", "add", "mul"}, reduction_mode))`, its a little cleaner.

Yeah that's better.